### PR TITLE
original datapoint DataSources, NationalClasses and OriginalData update 

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1126

### CODE REVIEW

File: client/store/pages/originalDataPoint/actions/getOriginalDataPoint.ts [hunk 1]
```diff
@@ -3,11 +3,12 @@ import { createAsyncThunk } from '@reduxjs/toolkit'
 import axios from 'axios'
 import { ApiEndPoint } from '@common/api/endpoint'
 import { OriginalDataPoint } from '@meta/assessment'
+import { addNationalClassPlaceHolder } from '@core/odp/odps/addNationalClassPlaceHolder'
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/getOriginalDataPoint.ts:6]: import { ODPs } from '@core/odp'

File: client/store/pages/originalDataPoint/actions/getOriginalDataPoint.ts [hunk 2]
```diff
@@ -3,11 +3,12 @@ import { createAsyncThunk } from '@reduxjs/toolkit'
...
 
 export const getOriginalDataPoint = createAsyncThunk<
   OriginalDataPoint,
   { assessmentName: string; cycleName: string; odpId: string }
 >('originalDataPoint/get/byId', async ({ assessmentName, cycleName, odpId }) => {
   const { data } = await axios.get(ApiEndPoint.Assessment.OriginalDataPoint.one(assessmentName, cycleName, odpId))
-  return data
+  return addNationalClassPlaceHolder(data)
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/getOriginalDataPoint.ts:13]: ODPs.addNationalClassPlaceHolder(data)

File: client/store/pages/originalDataPoint/actions/updateOriginalDataPoint.ts [hunk 3]
```diff
@@ -3,13 +3,15 @@ import { createAsyncThunk } from '@reduxjs/toolkit'
 import axios from 'axios'
 import { ApiEndPoint } from '@common/api/endpoint'
 import { OriginalDataPoint } from '@meta/assessment'
+import { addNationalClassPlaceHolder } from '@core/odp/odps/addNationalClassPlaceHolder'
+import { removeNationalClassPlaceHolder } from '@core/odp/odps/removeNationalClassPlaceHolder'
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/updateOriginalDataPoint.ts:7]: import { ODPs } from '@core/odp'

File: client/utils/copyPasteUtil.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as R from 'ramda'
```
Comment (minotogna) [inline client/utils/copyPasteUtil.ts:2]: can u remove ramda ?

**Review by minotogna — CHANGES_REQUESTED** [review]
do not use core.
move ODPs, all you need under meta and import them from '@meta/assessment'

File: client/utils/copyPasteUtil.ts (hunk 4 above)
Comment (rfontanarosa) [reply client/utils/copyPasteUtil.ts:2]: done

File: client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx [hunk 5]
```diff
@@ -6,14 +6,18 @@ import VerticallyGrowingTextField from '@client/components/VerticallyGrowingText
...
 import { useTranslation } from 'react-i18next'
 // import { useCountryIso } from '@client/hooks'
-// import { useAppDispatch } from '@client/store'
-import { useOriginalDataPoint } from '@client/store/pages/originalDataPoint'
+import { useAppDispatch } from '@client/store'
+import { useAssessment, useCycle } from '@client/store/assessment'
+import { OriginalDataPoint } from '@meta/assessment/originalDataPoint'
+import { OriginalDataPointActions, useOriginalDataPoint } from '@client/store/pages/originalDataPoint'
+import { deleteNationalClass } from '@meta/assessment/originalDataPoint/odps/deleteNationalClass'
+import { updateNationalClass } from '@meta/assessment/originalDataPoint/odps/updateNationalClass'
```
Comment (minotogna) [inline client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx:14]: use import ODPs 
and use ODPs.deleteNationalClass and ODPs.updateNationalClass
do not use individual methods

File: client/store/pages/originalDataPoint/actions/pasteNationalClass.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    { dispatch }
+  ) => {
+    const rawPastedData = readPasteClipboard(event, 'string')
+    const { updatedOdp } = handlePaste(columns, allowedClass, odp, allowGrow, rawPastedData, rowIndex, colIndex)
+    dispatch(
+      updateOriginalDataPoint({
+        assessmentName,
+        cycleName,
+        odpId: String(updatedOdp.id),
+        originalDataPoint: updatedOdp,
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/pasteNationalClass.ts:45]: why passing both, odpId and odp ? doesn't the odp have the id already ?

**Review by minotogna — CHANGES_REQUESTED** [review]

File: client/store/pages/originalDataPoint/actions/pasteNationalClass.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { ODPNationalClass } from '@core/odp'
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import { readPasteClipboard } from '@client/utils/copyPasteUtil'
+import { OriginalDataPoint } from '@meta/assessment'
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/pasteNationalClass.ts:5]: do not use core
move ODPNationalClass in '@meta/assessment'

File: client/store/pages/originalDataPoint/actions/updateNationalClass.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import { ODPNationalClass, ODPs } from '@core/odp'
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/updateNationalClass.ts:1]: move these in '@meta/assessment'

File: client/store/pages/originalDataPoint/actions/updateOriginalDataPoint.ts [hunk 9]
```diff
@@ -3,13 +3,14 @@ import { createAsyncThunk } from '@reduxjs/toolkit'
 import axios from 'axios'
 import { ApiEndPoint } from '@common/api/endpoint'
 import { OriginalDataPoint } from '@meta/assessment'
+import { ODPs } from '@meta/assessment/originalDataPoint'
```
Comment (minotogna) [inline client/store/pages/originalDataPoint/actions/updateOriginalDataPoint.ts:6]: export from '@meta/assessment' all ODP related objects !

**Review by minotogna — CHANGES_REQUESTED** [review]

File: client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx [hunk 10]
```diff
@@ -6,14 +6,17 @@ import VerticallyGrowingTextField from '@client/components/VerticallyGrowingText
 // import ReviewIndicator from '@webapp/app/assessment/components/review/reviewIndicator'
 import { useTranslation } from 'react-i18next'
 // import { useCountryIso } from '@client/hooks'
-// import { useAppDispatch } from '@client/store'
-import { useOriginalDataPoint } from '@client/store/pages/originalDataPoint'
+import { useAppDispatch } from '@client/store'
+import { useAssessment, useCycle } from '@client/store/assessment'
+import { OriginalDataPoint } from '@meta/assessment/originalDataPoint'
```
Comment (minotogna) [inline client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx:11]: export OriginalDataPoint from '@meta/assessment' !!!!!!!!!!!!!!

File: client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx [hunk 11]
```diff
@@ -24,17 +27,31 @@ const NationalClass: React.FC<Props> = (props) => {
...
   const classNameRowComments = useNationalClassNameComments(target)
   const validation = useNationalClassValidation(index)
 
+  const updateOriginalDataPoint = (originalDataPointUpdate: OriginalDataPoint) => {
+    dispatch(
+      OriginalDataPointActions.updateOriginalDataPoint({
+        cycleName: cycle.name,
+        assessmentName: assessment.props.name,
+        originalDataPoint: originalDataPointUpdate,
+        odpId: String(originalDataPointUpdate.id),
```
Comment (minotogna) [inline client/pages/OriginalDataPoint/components/NationalClasses/NationalClass/NationalClass.tsx:50]: why passing odpId and originalDataPoint ??????

**Review by minotogna — CHANGES_REQUESTED** [review]
please, please please please please , see my comments again

File: core/odp/odps/addNationalClassPlaceHolder.ts [hunk 12]
```diff
@@ -1,6 +1,7 @@
-import { ODP, ODPNationalClassFactory } from '@core/odp'
+import { ODPNationalClassFactory } from '@core/odp/odpNationalClass'
+import { OriginalDataPoint } from '@meta/assessment'
```
Comment (minotogna) [inline core/odp/odps/addNationalClassPlaceHolder.ts:2]: export ODPNationalClassFactory from '@meta/assessment'

Comment (minotogna) [reply core/odp/odps/addNationalClassPlaceHolder.ts:2]: core is deprecated !!!

**Review by minotogna — CHANGES_REQUESTED** [review]

File: core/odp/odps/addNationalClassPlaceHolder.ts (hunk 12 above)
Comment (rfontanarosa) [reply core/odp/odps/addNationalClassPlaceHolder.ts:2]: ok, what about this? import { Objects } from '@core/utils'

Comment (rfontanarosa) [reply core/odp/odps/addNationalClassPlaceHolder.ts:2]: this is core, so I should leave things as they were

Comment (minotogna) [reply core/odp/odps/addNationalClassPlaceHolder.ts:2]: we said in the meeting, utils will be moved out ;)

**Review by minotogna — APPROVED** [review]

**Pull request merged**